Source/UI/IdentityUI.swift

```diff
@@ -274,21 +274,23 @@ public class IdentityUI {
     public func presentIdentityProcess(from viewController: UIViewController, route: Route) {
         self.configuration.tracker?.loginMethod = route.loginMethod
         self.start(input: .byRoute(route, presentingViewController: viewController)) { [weak self] output in
-            self?.complete(with: output)
+            self?.complete(with: output, presentingViewController: viewController)
         }
     }
 
-    private func complete(with output: Output) {
+    private func complete(with output: Output, presentingViewController: UIViewController?) {
         guard IdentityUI.presentedIdentityUI != nil else {
             // IdentityUI has been already dismissed.
             return
         }
 
         let uiResult: IdentityUIResult?
+        var shouldPersistUser = false
         switch output {
-        case let .success(user):
+        case let .success(user, persistUser):
             self.configuration.tracker?.engagement(.network(.done))
             uiResult = .completed(user)
+            shouldPersistUser = persistUser
```

Comment:
Why can we not just store the tokens here? Could the user still cancel the login process because of requiredFields or something? Maybe its that you are fixing now?

Reply:
The patch gives a chance to client to do one last thing before finalizing. So they can choose to reset the flow, and then ya, not persisting.